src/date-input/date-input.tsx

```diff
@@ -46,12 +46,9 @@ export const DateInput = ({
     const nodeRef = useRef<HTMLDivElement>(null);
     const calendarRef = useRef<InternalCalendarRef>();
     const inputRef = useRef<StandaloneDateInputRef>();
-    const isMobile = useMediaQuery({
-        maxWidth: MediaWidths.mobileL,
-    });
 
     // show button if it is mobile view
-    const withButton = _withButton || isMobile;
+    const withButton = _withButton;
```

Comment:
since it's actually the same value, we can remove the prop renaming and use `withButton` directly